Home Screen Improvements: Add devices, activity log, and application summary

Pezmc opened the pull request:
## Description

Draft PR open for early implementation review, UI review, and for testing on staging.

Notable Changes:
 - Includes three devices & instances with most recent activity log entries
 - Includes the most recent activity log entry
 - Incudes the total devices/instances/deviceGroups/snapshots/pipelines
 - Adds a link to the device editor (if it is enabled)

![Screenshot 2024-02-01 at 16 30 36](https://github.com/FlowFuse/flowfuse/assets/507155/50d5bbf5-52c0-4d0b-bc21-81855784acb1)

Open tasks:
 - [x] Test coverage of devices, API changes and editor button
 - [ ] Test coverage of the UI - Moving to a follow up
 - [ ] Adding icons instead of text for the totals - Moving to a follow up

## Related Issue(s)

Fix #3398

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

Pezmc commented on forge/db/models/Application.js:
```diff
@@ -91,12 +92,129 @@ module.exports = {
+                                        ORDER BY "createdAt" DESC
+                                        LIMIT 1
+                                    )`),
+                                    'mostRecentAuditLogCreatedAt'
```
I trialed left joining the AuditLog table, however I was unable to get Sequelize to run it in the same query, thus is was not possible to order the instances and devices by `mostRecentAuditLogCreatedAt`, see: https://github.com/FlowFuse/flowfuse/compare/feat-3398-home-screen...feat-3398-home-screen-left-join

knolleary commented:
Quick test locally looks good - including a couple devices. Happy to push to staging for further verification.

knolleary commented:
Approved pending the failing tests being resolved.

Pezmc commented:
I've rebased away all of the commits getting the tests to behave on github and squashed them down into [07383f4](https://github.com/FlowFuse/flowfuse/pull/3427/commits/07383f416e50cdd3c03df650fd4ac822d0eb4db1).
